lib/parser.js

```diff
@@ -202,15 +207,39 @@ async function customDocumentOperations(parsedJSON, asyncapiYAMLorJSON, initialF
   validateChannels(parsedJSON, asyncapiYAMLorJSON, initialFormat);
   validateOperationId(parsedJSON, asyncapiYAMLorJSON, initialFormat, OPERATIONS);
 
-  await customComponentsMsgOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options);
-  await customChannelsOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options);
+  await customComponentsMsgOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options, schemaValidationErrors);
+  await customChannelsOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options, schemaValidationErrors);
 }
 
-async function validateAndConvertMessage(msg, originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload) {
+async function validateAndConvertMessage(msg,name ,originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload, options, schemaValidationErrors) {
   //check if the message has been parsed before
   if (xParserMessageParsed in msg && msg[String(xParserMessageParsed)] === true) return;
   const defaultSchemaFormat = getDefaultSchemaFormat(parsedAsyncAPIDocument.asyncapi);
   const schemaFormat = msg.schemaFormat || defaultSchemaFormat;
+  const parser = PARSERS[String(schemaFormat)];
```

Comment:
```suggestion
  const customSchemaParser = PARSERS[String(schemaFormat)];
```

I think more clear name